### PR TITLE
Poker drawer discussion thread

### PR DESCRIPTION
**Pull request by nickoferrall**
PR resolves issue #4260

### Updated AC

- [ ]  drawer looks like the figma spec: https://www.figma.com/file/csERjrh67KZKgnT5VahFSb/Sprint-Poker?node-id=2689%3A118425
- [ ] drawer is swipable on mobile
- [ ] able to open & close drawer on desktop
- [ ] discussion thread is functional

### Additional consideration

- Added a badge icon if there's an unread comment

### CODE REVIEW

**Comment by nickoferrall** [comment]
Hey @ackernaut, it'd be great to get your thoughts on the design on this one! I've added the badge if there's an unread comment - I based the design off of [these examples](http://localhost:3000/meet/q-pGU8e9xk/estimate/2)

**Comment by ackernaut** [comment]
Good deal! Have you tried merging the latest from master? I’m wondering if there are conflicts.

**Comment by ackernaut** [comment]
The ‘these examples’ link is pointing to a localhost URL

**Comment by nickoferrall** [comment]
🤣  that's what happens when linking late on a Friday. These MUI designs: https://material-ui.com/components/badges/

Yep, I merged the latest from master and fixed the conflicts

**Comment by ackernaut** [comment]
@nickoferrall this is looking good, a few things to note:
- [ ] It seems the thread doesn’t show any comments for a second user (in my case a participant joining in another browser window locally). The 2nd user can add comments which the 1st user (for me, the facilitator) sees. The 1st user can see and add comments as expected.
- [ ] Check out my styles here #4365 and merge if you accept
- [ ] For the unread logic, it seems it sets the state of the thread as having unread comments if I load in a mobile viewport even though I had already seen those comments. I know the logic is simple and I’d like to err on the side of less noise in this case. So if I load in a mobile viewport (or any viewport size) I think we should show the alert badge only when comments have been added after my session has started (and the drawer is closed, of course) — feel free to push back on this opinion

**Review by ackernaut — CHANGES_REQUESTED** [review]
See my checklist, at a minimum the thread needs to be debugged for a second viewer

**Comment by nickoferrall** [comment]
Thanks for the review @ackernaut. I've updated the unread comment logic, merged your PR and made sure that the card doesn't hide behind the sidebar.

The discussion thread is functional for me for both user 1 & user 2, i.e. both users can see each others' comments. If this still isn't working for you, let me know and I'll investigate further.

If you're happy with the updates, I'll request a final review from Matt to check out the code

**Comment by ackernaut** [comment]
Well, the sad news is I ended my existing poker meeting and started another. I added 3 stories and can’t seem to hit the estimate URL. I’m not sure if this is a regression (or from where?, e.g. from master) or something local:
- [ ] Start a poker meeting for 2 users
- [ ] Select stories from Jira
- [ ] Advance to the estimate phase but can’t

I’ll try troubleshooting locally:
```
Error: Not Found
    at eval (webpack-internal:///./packages/server/dataloader/customLoaderMakers.ts:342:81)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async promiseAllPartial (webpack-internal:///./packages/client/utils/promiseAllPartial.ts:11:19) { cloudId: '10066', issueKey: undefined, teamId: 'RRGNAyipDK' }
SEND TO SENTRY Error: Not Found
    at eval (webpack-internal:///./packages/server/dataloader/customLoaderMakers.ts:342:81)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async promiseAllPartial (webpack-internal:///./packages/client/utils/promiseAllPartial.ts:11:19) { cloudId: '10064', issueKey: undefined, teamId: 'RRGNAyipDK' }
SEND TO SENTRY Error: Not Found
    at eval (webpack-internal:///./packages/server/dataloader/customLoaderMakers.ts:342:81)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async promiseAllPartial (webpack-internal:///./packages/client/utils/promiseAllPartial.ts:11:19) { cloudId: '10063', issueKey: undefined, teamId: 'RRGNAyipDK' }
Trace: {
  error: '[{"message":"fields.find is not a function","locations":[{"line":631,"column":3},{"line":617,"column":3},{"line":1033,"column":3}],"path":["viewer","meeting","phases",1,"stages",4,"serviceField"]},{"message":"fields.find is not a function","locations":[{"line":631,"column":3},{"line":617,"column":3},{"line":1033,"column":3}],"path":["viewer","meeting","phases",1,"stages",2,"serviceField"]},{"message":"fields.find is not a function","locations":[{"line":631,"column":3},{"line":617,"column":3},{"line":1033,"column":3}],"path":["viewer","meeting","phases",1,"stages",0,"serviceField"]}]'
}
    at executeGraphQL (webpack-internal:///./packages/server/graphql/executeGraphQL.ts:107:13)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Redis.onMessage (webpack-internal:///./packages/gql-executor/gqlExecutor.ts:15:18)
SEND TO SENTRY [Error: fields.find is not a function] undefined
SEND TO SENTRY Error: Rate limit reached
    at Object.eval [as MutationsetAppLocationResolver] (webpack-internal:///./packages/server/graphql/rateLimit.ts:35:92)
    at Array.setAppLocationMutationsetAppLocationResolverMutation (eval at compileQuery (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/dist/execution.js:69:56), <anonymous>:140:39)
    at query (eval at compileQuery (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/dist/execution.js:69:56), <anonymous>:20:25)
    at Object.SetAppLocationMutation [as query] (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/dist/execution.js:135:33)
    at executeGraphQL (webpack-internal:///./packages/server/graphql/executeGraphQL.ts:97:40)
    at Redis.onMessage (webpack-internal:///./packages/gql-executor/gqlExecutor.ts:15:24)
    at Redis.emit (events.js:314:20)
    at Redis.EventEmitter.emit (domain.js:486:12)
    at DataHandler.handleSubscriberReply (/Users/terryacker/code/parabolinc/parabol/node_modules/ioredis/built/DataHandler.js:80:32)
    at DataHandler.returnReply (/Users/terryacker/code/parabolinc/parabol/node_modules/ioredis/built/DataHandler.js:47:18)
    at JavascriptRedisParser.returnReply (/Users/terryacker/code/parabolinc/parabol/node_modules/ioredis/built/DataHandler.js:21:22)
    at JavascriptRedisParser.execute (/Users/terryacker/code/parabolinc/parabol/node_modules/redis-parser/lib/parser.js:544:14)
    at Socket.<anonymous> (/Users/terryacker/code/parabolinc/parabol/node_modules/ioredis/built/DataHandler.js:25:20)
    at Socket.emit (events.js:314:20)
    at Socket.EventEmitter.emit (domain.js:486:12)
    at addChunk (_stream_readable.js:308:12) {
  query: 'setAppLocation',
  variables: '{"location":"/meet/nms17FYWb6"}'
}
SEND TO SENTRY Error: Rate limit reached
    at Object.eval [as MutationsetAppLocationResolver] (webpack-internal:///./packages/server/graphql/rateLimit.ts:35:92)
    at Array.setAppLocationMutationsetAppLocationResolverMutation (eval at compileQuery (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/dist/execution.js:69:56), <anonymous>:140:39)
    at query (eval at compileQuery (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/dist/execution.js:69:56), <anonymous>:20:25)
    at Object.SetAppLocationMutation [as query] (/Users/terryacker/code/parabolinc/parabol/node_modules/graphql-jit/dist/execution.js:135:33)
    at executeGraphQL (webpack-internal:///./packages/server/graphql/executeGraphQL.ts:97:40)
    at Redis.onMessage (webpack-internal:///./packages/gql-executor/gqlExecutor.ts:15:24)
    at Redis.emit (events.js:314:20)
    at Redis.EventEmitter.emit (domain.js:486:12)
    at DataHandler.handleSubscriberReply (/Users/terryacker/code/parabolinc/parabol/node_modules/ioredis/built/DataHandler.js:80:32)
    at DataHandler.returnReply (/Users/terryacker/code/parabolinc/parabol/node_modules/ioredis/built/DataHandler.js:47:18)
    at JavascriptRedisParser.returnReply (/Users/terryacker/code/parabolinc/parabol/node_modules/ioredis/built/DataHandler.js:21:22)
    at JavascriptRedisParser.execute (/Users/terryacker/code/parabolinc/parabol/node_modules/redis-parser/lib/parser.js:544:14)
    at Socket.<anonymous> (/Users/terryacker/code/parabolinc/parabol/node_modules/ioredis/built/DataHandler.js:25:20)
    at Socket.emit (events.js:314:20)
    at Socket.EventEmitter.emit (domain.js:486:12)
    at addChunk (_stream_readable.js:308:12) { query: 'setAppLocation', variables: '{"location":null}' }
Trace: {
  error: '[{"message":"429 Too Many Requests","locations":[{"line":4,"column":3}],"path":["setAppLocation"]}]'
}```

**Comment by ackernaut** [comment]
This is suspect: `fields.find is not a function`

**Comment by ackernaut** [comment]
Ah, seems I’m getting the same issue when on the `master` branch.

**Comment by ackernaut** [comment]
- [x] drawer looks like the figma spec: https://www.figma.com/file/csERjrh67KZKgnT5VahFSb/Sprint-Poker?node-id=2689%3A118425
- [x] drawer is swipable on mobile
- [x] able to open & close drawer on desktop
- [x] discussion thread is functional

This is looking good. I like the updated logic for unread comments. The thread worked for 2 players.

**Review by ackernaut — APPROVED** [review]

File: packages/client/mutations/AddCommentMutation.ts [hunk 1]
```diff
@@ -63,6 +69,8 @@ const AddCommentMutation: StandardMutation<TAddCommentMutation> = (
     variables,
     updater: (store) => {
       const payload = store.getRootField('addComment')
+      const {meetingId} = variables
+      payload.setValue(meetingId, 'meetingId')
```
Comment (mattkrick) [inline packages/client/mutations/AddCommentMutation.ts:73]: this pattern is a little too dangerous. Remember, the updater function is shared by mutations & subscriptions because we guarantee that they'll have the same payload. if we break that contract, then subscriptions can break.

if you need a meetingId, it's best for the server to just return it

File: packages/client/mutations/AddCommentMutation.ts [hunk 2]
```diff
@@ -38,6 +38,12 @@ export const addCommentMeetingUpdater: SharedUpdater<AddCommentMutation_meeting>
   payload,
   {store}
 ) => {
+  const meetingId = payload.getValue('meetingId') as string
```
Comment (mattkrick) [inline packages/client/mutations/AddCommentMutation.ts:41]: how does this work for subscriptions?

File: packages/server/graphql/intranetGraphQLHandler.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import uWSAsyncHandler from './uWSAsyncHandler'
 
 interface IntranetPayload {
   query: string
-  variables: object
+  variables: any
```
Comment (mattkrick) [inline packages/server/graphql/intranetGraphQLHandler.ts:12]: curious why the change? any includes a primitive, which could break it

File: packages/server/utils/getJiraCloudIdAndKey.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const getJiraCloudIdAndKey = (id: string): [cloudId: string, issueKey: string] => {
```
Comment (mattkrick) [inline packages/server/utils/getJiraCloudIdAndKey.ts:1]: i moved this one to /client/utils since we're using it in both no, we could probably just use that one?

**Review by mattkrick — CHANGES_REQUESTED** [review]
good stuff! just a couple bugs, after those are fixed feel free to merge

File: packages/client/mutations/AddCommentMutation.ts (hunk 1 above)
Comment (nickoferrall) [reply packages/client/mutations/AddCommentMutation.ts]: Thanks, that's helpful to know. I've added the `meetingId` to the payload

File: packages/server/graphql/intranetGraphQLHandler.ts (hunk 3 above)
Comment (nickoferrall) [reply packages/server/graphql/intranetGraphQLHandler.ts]: Eslint gave the following error when using `object` type:
```
parabol-server: 12:14  error  Don't use `object` as a type. The `object` type is currently hard to use ([see this issue]((https://github.com/microsoft/TypeScript/issues/21732))).
parabol-server: Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys  @typescript-eslint/ban-types
```
I haven't been able to log the variables to see exactly what they are so I've updated `any` to be the suggested `Record<string, unknown>`, making sure that variables can't be primitives while still giving us flexibility in the object with `unknown`
